### PR TITLE
EncodeRLP for Amount

### PR DESCRIPTION
**Pull request by kfangw**

### CODE REVIEW

**Comment by kfangw** [comment]
To patch this,
Every integration test cases should be fixed
And the clients should be fixed also.

**Comment by kfangw** [comment]
please wait. I have to adjust the integration tests
Calculcate every signatures.....

File: lib/common/amount.go [hunk 1]
```diff
@@ -42,6 +43,11 @@ func (this Amount) Invariant() {
 	}
 }
 
+func (a Amount) EncodeRLP(w io.Writer) (err error) {
+	w.Write([]byte(a.String()))
```
Comment (zzim2x) [inline lib/common/amount.go]:
```
w.Write(a.String())
```

Comment (zzim2x) [reply lib/common/amount.go]: "10000"
actual : 0x853130303030
got : 0x3130303030

[the remark this answers is not in the document]
Comment (kfangw) [reply lib/common/amount.go]: What you mean by big? BigEndian?
Where to report?

[the remark this answers is not in the document]
Comment (zzim2x) [reply lib/common/amount.go]: https://github.com/bosnet/sebak/issues/733

Sorry.
big -> bug

[the remark this answers is not in the document]
Comment (anarcher) [reply lib/common/amount.go]: `EncodeRLP` is a part of `rlp.Encoder` interface.
It means that the `EncodeRLP` method should be encoding RLP itself.

```
bs,err := rlp.EncodeToBytes(a.String())
if err != nil { panic(err) }
w.Write(bs)
```
- https://golang.org/pkg/encoding/json/#Marshaler

[the remark this answers is not in the document]
Comment (anarcher) [reply lib/common/amount.go]: Oh , `rlp.Encode(w,a.String())` is better then `rlp.EncodeToBytes(...)` :->

**Comment by Geod24** [comment]
> To encode Amount as string(byte array).

Why do we encode an `Amount` as a string???

**Pull request merged**